### PR TITLE
Makefile/Docs Edits

### PR DESCRIPTION
**Pull request by jkprow**
add Makefile compatibility with test process
add CONTRIBUTING.md with instructions for adding code
update README.md

### CODE REVIEW

**Comment by jkprow** [comment]
retest this please

File: Makefile [hunk 1]
```diff
@@ -17,18 +17,18 @@ clean: db_drop
...
 	@rm -rf node_modules client/public/vendor && \
 	npm cache clear >>log/cleanup.log 2>&1 && \
-	bower cache clean >>log/cleanup.log 2>&1 && ([ $$? -eq 0 ] && echo "success!") || echo "failure!"
+	bower cache clean --allow-root >>log/cleanup.log 2>&1 && ([ $$? -eq 0 ] && echo "success!") || echo "failure!"
 
 db_build: ;@echo "Building database and logging to log/db.log...";
 	@echo `date` > log/db.log
 	@echo "Creating database..." | tee -a log/db.log
-	@psql -h "$(db)" -f scripts/createDatabase.sql >>log/db.log 2>&1 && ([ $$? -eq 0 ] && echo "success!") || echo "failure!" | tee -a log/start.log
+	@psql -U postgres -h "$(db)" -f scripts/createDatabase.sql >>log/db.log 2>&1 && ([ $$? -eq 0 ] && echo "success!") || echo "failure!"
```
Comment (alexburner) [inline Makefile]: Is this still using that db=<hostname_of_postgres_server> thing? Could you pop a quick comment in there about that? The old makefile had "run as: make start db=<hostname_of_postgres_server>"

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
-fresh: cleanup install start
+fresh: cleanup install
```
Comment (alexburner) [inline Makefile]: this is awesome, great idea

File: Makefile (hunk 1 above)
Comment (alexburner) [inline Makefile]: NVM this is now option for plebs like me

**Pull request merged**